src/system_info/system_info_network_mobile.cc

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2013 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "system_info/system_info_network.h"
+
+SysInfoNetwork::SysInfoNetwork()
+    : type_(SYSTEM_INFO_NETWORK_UNKNOWN) {}
+
+SysInfoNetwork::~SysInfoNetwork() {}
+
+void SysInfoNetwork::StartListening() {
+  vconf_notify_key_changed(VCONFKEY_NETWORK_STATUS,
+      (vconf_callback_fn)OnNetworkTypeChanged, this);
+
+  vconf_notify_key_changed(VCONFKEY_TELEPHONY_SVCTYPE,
+      (vconf_callback_fn)OnNetworkTypeChanged, this);
```

Comment:
Here, Why you register the same callback function? Maybe you need  OnCellularNetworkTypeChaged.